modules/machineset-yaml-alibaba.adoc

```diff
@@ -0,0 +1,176 @@
+            - Key: Name
+              Value: <infrastructure_id>-sg-<role>
+            type: Tags
+          systemDisk: <10>
+            category: cloud_essd
+            size: <disk_size>
+          tag: <9>
+          - Key: kubernetes.io/cluster/<infrastructure_id>
+            Value: owned
+          - Key: GISV
+            Value: ocp
+          - Key: sigs.k8s.io/cloud-provider-alibaba/origin
+            Value: ocp
+          userDataSecret:
+            name: <user_data_secret> <11>
+          vSwitch:
+            tags: <9>
+            - Key: kubernetes.io/cluster/<infrastructure_id>
+              Value: owned
+            - Key: GISV
+              Value: ocp
+            - Key: sigs.k8s.io/cloud-provider-alibaba/origin
+              Value: ocp
+            - Key: Name
+              Value: <infrastructure_id>-vswitch-<region>-<zone>
+            type: Tags
+          vpcId: ""
+          zoneId: <region>-<zone> <12>
+ifdef::infra[]
+        taints: <13>
+        - key: node-role.kubernetes.io/infra
+          effect: NoSchedule
+endif::infra[]
+----
+<1> Specify the infrastructure ID that is based on the cluster ID that you set when you provisioned the cluster. If you have the OpenShift CLI installed, you can obtain the infrastructure ID by running the following command:
++
+[source,terminal]
+----
+$ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
+----
+ifndef::infra[]
+<2> Specify the node label to add.
+<3> Specify the infrastructure ID, node label, and region.
+endif::infra[]
+ifdef::infra[]
+<2> Specify the `<infra>` node label.
+<3> Specify the infrastructure ID, `<infra>` node label, and region.
+endif::infra[]
+<4> Specify the image to use. Note: Get an image from an existing default machine set for the cluster.
+<5> Specify the instance type you want to use for the machine set.
+<6> Specify the name of the RAM role to use for the machine set .
+// Should they use the `ramRoleName` value that the installer populates in the default machine set?
+<7> Specify the region to place machines on.
+<8> Specify the resource group for the cluster. You can use the value in the machine set that that the installer creates, or specify a different one.
+<9> Specify the tags to use for the machine set. Minimally, you must include the tags that the installer populates in the default machine set it creates. You can include additional tags as needed.
+// Would it be better to leave the information here more vague so that the user doesn't get hung up on specifics? For instance, replace `kubernetes.io/cluster/<infrastructure_id>` with `<tag_value_1>`?
+<10> Specify the type and size of the root disk. Use the `category` value that the installer populates in the default machine set it creates. If required, specify a different value in gigabytes for `size`.
+<11> Specify the name of the secret in the user data YAML file that is in the `openshift-machine-api` namespace.
+////
+Is this optional, or required?
```

Comment:
Question for devs.

Reply:
Required. This secret defines, alongside the CA certs, the MCS URL + MCP name[1] which will serve the rendered MachineConfigs.
I would keep the value the same as default MachineSets for workers, but it could be customized when creating a new MCP (not sure if it is the target for this doc).

Looking to [AWS doc](https://docs.openshift.com/container-platform/4.9/machine_management/creating-infrastructure-machinesets.html#machineset-yaml-aws_creating-infrastructure-machinesets), it points to worker one:
```
          userDataSecret:
            name: worker-user-data
```

[1] example of Machine config server endpoint for `worker` MCP
`$ oc get secrets -n openshift-machine-api worker-user-data -o json |jq -r .data.userData |base64 -d |jq .ignition.config.merge`
```json
[
  {
    "source": "https://api-int.lab.devcluster.openshift.com:22623/config/worker"
  }
]
```

Reply:
Ok, thank you. This doc should be focused on the basic task of adding machine sets to expand the cluster, so I think we can leave it at "use the same thing the installer populates" :)